models/url_pydantic_models.py

```diff
@@ -1,12 +1,16 @@
 from pydantic import BaseModel
+from datetime import datetime
+from typing import Optional
 # Define your models here
 
 
 class URLRequest(BaseModel):
     url: str
```

Comment:
Update the url field in the URLRequest model to use Pydantic’s HttpUrl type instead of str. This will automatically validate that only properly formatted URLs (e.g., starting with http:// or https://) are accepted.